prow/config/githuboauth.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"encoding/base64"
+	"encoding/gob"
+
+	"github.com/gorilla/sessions"
+	"golang.org/x/oauth2"
+)
+
+type Cookie struct {
+	Secret string `json:"secret,omitempty"`
+}
+
+// GithubOAuthConfig is a config for requesting users access tokens from Github API. It also has
+// a Cookie Store that retains user credentials deriving from Github API.
+type GithubOAuthConfig struct {
+	ClientID         string   `json:"client_id"`
+	ClientSecret     string   `json:"client_secret"`
+	RedirectURL      string   `json:"redirect_url"`
+	Scopes           []string `json:"scopes,omitempty"`
+	FinalRedirectURL string   `json:"final_redirect_url"`
+
+	GithubTokenSession string `json:"token_session"`
+	GithubTokenKey     string `json:"token_key"`
```

Comment:
Why are these two fields configurable unlike `oauthSessionCookie` and `stateKey`? It looks like they could be constant key values?

Reply:
These fields are used among endpoints. I want to keep these values constant so I make them "configurable". WDYT?

Reply:
I don't follow, if you want to keep the values constant you would **not** want them to be configurable, right?  These fields seem to be used in a similar fashion to the `oauthSessionCookie` and `stateKey` fields which are constants and not configurable.

Reply:
done